src/common.hpp

```diff
@@ -554,4 +554,35 @@ class Stopwatch
     std::chrono::high_resolution_clock::time_point tstart;
 };
 
+////////////////////////////////////////////////////////////////////////////////
+// A global log that can stream to an external source (like unreal).
+// Intended to replace calls to printf when stdout is not available.
+// Not thread safe or intended for speed.
+////////////////////////////////////////////////////////////////////////////////
+
+static void (*EXTERNAL_GLOBAL_LOG)(const char* message) = nullptr;
+
+#ifdef BHC_CPU
+inline void GlobalLog(const char* message, ...) {
+	va_list argp;
+	va_start(argp, message);
+	if (EXTERNAL_GLOBAL_LOG == nullptr) {
+		vprintf(message, argp);
+	}
+	else {
+		char* buffer = new char[strlen(message)];
```

Comment:
I noticed that this was never deallocated, but it's even worse than that. `strlen(message)` is the length of the format string, which is usually smaller than the final output string. This will overflow the buffer almost every time.

Reply:
Sigh, clearly my finest moment. I'm pulling the logging stuff.

Reply:
No need to pull it--I'm modifying it to do it a different way. The changes from `printf` to `GlobalLog` will stay.